Modules/Filtering/FastMarching/include/itkFastMarchingImageToNodePairContainerAdaptor.h

```diff
@@ -98,12 +98,13 @@ class ITK_TEMPLATE_EXPORT FastMarchingImageToNodePairContainerAdaptor : public O
     is a binary mask; thus null values are considered as
     FastMarchingTraitsBase::Forbidden points
 
-    \li else (m_IsForbiddenImageBinaryMask is \c fasle) non null values
+    \li else (m_IsForbiddenImageBinaryMask is \c false) non null values
     represents FastMarchingTraitsBase::Forbidden points*/
   void
   SetForbiddenImage(const ImageType * iImage);
 
   itkSetMacro(IsForbiddenImageBinaryMask, bool);
+  itkGetConstMacro(IsForbiddenImageBinaryMask, bool);
```

Comment:
There is some inconsistency across the code base concerning the boolean getter macros: `itkGetConstReferenceMacro`, `itkGetConstMacro` and `itkGetMacro` all three are used. I guess we prefer one of the first two, but I'm not sure which of the two should be consistently used. Thoughts are welcome.

Reply:
I think that const is preferred for simple types (int, bool etc) and const reference for others (std::string, itk::image etc). If modification is required, then `itkGetModifiableObjectMacro` should be used. And `itkGetMacro` should be replaced by one of the other 3, as appropriate.

Reply:
OK. I'll try to put this to the documentation of the macros or the ITK SW Guide.

As always, thanks for the review @dzenanz :100:.

Reply:
> I think that const is preferred for simple types (int, bool etc) and const reference for others (std::string, itk::image etc).

With a little C++ acrobatics involving templates and/or SFINAE, we might be able to create a macro that figures out at compile time whether to return by value or by reference.